stripe/api_resources/abstract/api_resource.py

```diff
@@ -3,17 +3,19 @@
 from stripe import api_requestor, error, util
 from stripe.stripe_object import StripeObject
 from urllib.parse import quote_plus
-from typing import ClassVar
+from typing import ClassVar, Generic, TypeVar, cast
 
+T = TypeVar("T", bound="StripeObject")
 
-class APIResource(StripeObject):
+
+class APIResource(StripeObject, Generic[T]):
     OBJECT_NAME: ClassVar[str]
 
     @classmethod
-    def retrieve(cls, id, api_key=None, **params):
+    def retrieve(cls, id, api_key=None, **params) -> T:
```

Comment:
While we are in these files should we type other methods that return `T`? Like `refresh`?

Reply:
I want to put a little more thought into `refresh` and would rather it stay out of scope here.